sdk/tests/conformance/uniforms/gl-uniform-arrays.html

```diff
@@ -305,206 +288,9 @@
-          extraName: littleName,
-        });
-      }
-    }
-  }
-}
-
-var allRequiredUniformLocationsQueryable = true;
-function testUniformOptimizationIssues(testIndex) {
-  var test = tests[testIndex];
-  debug("");
-  debug(test.desc);
-  var fSrc = test.source;
-  if (!fSrc) {
-    fSrc = wtu.replaceParams(wtu.getScript(test.shader), test);
-  }
-
-  var consoleElem = document.getElementById("console");
-  wtu.addShaderSource(
-      consoleElem, "vertex shader", vSrc);
-  wtu.addShaderSource(
-      consoleElem, "fragment shader", fSrc);
-
-  var program = wtu.loadProgram(gl, vSrc, fSrc);
-  gl.useProgram(program);
-
-  var colorbLocation = gl.getUniformLocation(program, test.extraName + "[0]");
-  if (colorbLocation) {
-    gl.uniform4fv(colorbLocation, [0, 1, 0, 0]);
-  }
-
-  // Ensure that requesting an array uniform past MaxInt32PlusOne returns no uniform
-  var nameMaxInt32PlusOne = test.arrayName + "[" + (test.usedUniformVector + MaxInt32PlusOne) + "]";
-  assertMsg(gl.getUniformLocation(program, nameMaxInt32PlusOne) === null,
-    "Requesting " + nameMaxInt32PlusOne + " uniform should return a null uniform location");
-
-  // Set just the used uniform
-  var name = test.arrayName + "[" + test.usedUniformVector + "]";
-  var uniformLocation = gl.getUniformLocation(program, name);
-  gl.uniform4fv(uniformLocation, test.color);
-  wtu.setupIndexedQuad(gl, 1);
-  wtu.clearAndDrawIndexedQuad(gl, 1);
-  wtu.checkCanvas(gl, [0, 255, 0, 255], "should be green");
-
-  // Set all the unused uniforms
-  var locations = [];
-  allRequiredUniformLocationsQueryable = true;
-  for (var ii = 0; ii < test.maxUniformVectors; ++ii) {
-    var name = test.arrayName + "[" + ii + "]";
-    var uniformLocation = gl.getUniformLocation(program, name);
-    locations.push(uniformLocation);
-    if (ii == test.usedUniformVector) {
-      continue;
-    }
-    // Locations > usedUnformVector may not exist.
-    // Locations <= usedUniformVector MUST exist.
-    if (ii <= test.usedUniformVector && (uniformLocation === undefined || uniformLocation === null)) {
-      allRequiredUniformLocationsQueryable = false;
-    }
-    gl.uniform4fv(uniformLocation, [1, 0, 0, 1]);
-  }
-  shouldBeTrue("allRequiredUniformLocationsQueryable");
-  var positionLoc = gl.getAttribLocation(program, "a_position");
-  wtu.setupIndexedQuad(gl, 1, positionLoc);
-  wtu.clearAndDrawIndexedQuad(gl, 1);
-  wtu.checkCanvas(gl, [0, 255, 0, 255], "should be green");
-
-  // Check we can read & write each uniform.
-  // Note: uniforms past test.usedUniformVector might not exist.
-  for (var ii = 0; ii < test.maxUniformVectors; ++ii) {
-    gl.uniform4fv(locations[ii], [ii + 4, ii + 2, ii + 3, ii + 1]);
-  }
-
-  var kEpsilon = 0.01;
-  var isSame = function(v1, v2) {
-    return Math.abs(v1 - v2) < kEpsilon;
-  };
-
-  for (var ii = 0; ii < test.maxUniformVectors; ++ii) {
-    var location = locations[ii];
-    if (location) {
-      var value = gl.getUniform(program, locations[ii]);
```

Comment:
It takes about 10 seconds to finish this loop on a debug version of Chrome.
As the loop is totally executed for 18 times, it is the most time consuming part in this test.